repository_service_tuf/cli/key/info.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-FileCopyrightText: 2022-2023 VMware Inc
+#
+# SPDX-License-Identifier: MIT
+
+from rich import prompt, table
+from rich.console import Console  # type: ignore
+
+from repository_service_tuf.cli import click
+from repository_service_tuf.cli.key import key
+from repository_service_tuf.constants import KeyType
+from repository_service_tuf.helpers.tuf import load_key
+
+console = Console()
+
+
+@click.option(
+    "--show-private",
+    help=(
+        "Show the private key. WARNING: use private key information carefully."
+    ),
+    default=False,
+    show_default=True,
+    is_flag=True,
+)
+@key.command()
+def info(show_private: bool) -> None:
+    """Show key information"""
+
+    keytype: str = prompt.Prompt.ask(
+        "\nChoose key type",
+        choices=KeyType.get_all_members(),
+        default=KeyType.KEY_TYPE_ED25519.value,
+    )
+
+    filepath: str = prompt.Prompt.ask(
+        "Enter the key's [green]filename[/]",
+    )
```

Comment:
At least for the `keytype` (you can actually also get key id and public key, not private), I believe we can improve the user experience a bit and infer it from the public key file. Something like:
```suggestion
    filepath: str = prompt.Prompt.ask("Enter the key's [green]filename[/]")
    
    # TODO: Move imports to the top of the file
    import json
    import os
    # If the public key file exists, infer the key type from there.
    public_key_filename = filepath + ".pub"
    if os.path.isfile(public_key_filename) is True:
        with open(public_key_filename, mode="rb") as public_key_file:
            public_key: dict = json.load(public_key_file)
            keytype: str = public_key["keytype"]
    else:
        keytype: str = prompt.Prompt.ask(
            "\nChoose key type",
            choices=KeyType.get_all_members(),
            default=KeyType.KEY_TYPE_ED25519.value,
        )
```

Reply:
Unfortunately, we can't because rsa public keys created with the `rstuf key generate` command are stored as PEM, and not in the proprietary json format. See docstrings of used [`securesystemslib.interface.generate_and_write_*_keypair` functions](https://github.com/secure-systems-lab/securesystemslib/blob/main/securesystemslib/interface.py) for reference.

FYI: `securesystemslib.interface` is currently being re-designed. Until that I suggest to use it as is, without too much UX optimisation.

Reply:
Are you sure?
I tested yesterday with `rstuf key generate` and it created the .pub and private files. Then if the public file exists, it's simply a JSON structure that can be read.
I also tested the whole flow and it worked.

Reply:
From looking at the code, I'm pretty sure it doesn't work with RSA keys. But I didn't try it out. Did you test with RSA keys?

Reply:
Yep, just tried with rsa. The pub key is in PEM format. So your proposal doesn't work. You could of course try to json decode, and, if that fails, try to load as rsa. But as I said above, I wouldn't spend too much time on optimising the UX around `securesystemslib.interface`, as we're hopefully gonna have something better soon.

Reply:
Thank you folks!